ec2utils/ec2uploadimg/lib/ec2utils/ec2uploadimg.py

```diff
@@ -367,6 +377,22 @@ def _create_target_root_volume(self):
            the image we are creating."""
         return self._create_volume('%s' % self.root_volume_size)
 
+    # ---------------------------------------------------------------------
+    def _create_temporary_private_key(self, key_name='temporary_ec2_uploadkey'):
+        if self.verbose:
+            print('Creating temporary key pair')
+        suffix = ''.join(random.choices(string.ascii_uppercase + string.digits, k=12))
+        self.ssh_key_pair_name = key_name + '-' + suffix
+        self.ssh_key_private_key_file = os.path.join('/tmp/', self.ssh_key_pair_name + '.key')
+        secret_key_content = self._connect().create_key_pair(KeyName=self.ssh_key_pair_name)
+        if self.verbose:
+            print('Write temporary key to ', self.ssh_key_private_key_file)
+        file = open(self.ssh_key_private_key_file, 'w')
+        file.write(secret_key_content['KeyMaterial'])
```

Comment:
Here the content of the private key will be leaked to every user on the system. Because "open with 'w'" creates the fie with the default mask, which is generally 644, but the private key file needs to be 600. The file needs to be opened with 600 and should be kept in memory.

Reply:
Ouch, you are right :worried: